src/main/scala/fs2/Process1.scala

```diff
@@ -104,6 +104,12 @@ object process1 {
   def scan1[F[_],I](f: (I, I) => I)(implicit F: NotNothing[F]): Handle[F,I] => Pull[F,I,Handle[F,I]] =
     Pull.receive1 { case o #: h => scan(o)(f).apply(h) }
 
+  /** Writes the sum of all input elements, or zero if the input is empty. */
+  def sum[F[_],I:Numeric](implicit F: NotNothing[F]): Handle[F,I] => Pull[F,I,Handle[F,I]] = {
```

Comment:
These are reasonable points. I'll give it some more thought, but am honestly still leaning towards no. I just consider providing `Monoid` to be out of scope for this library, even if it adds a bit of convenience.

Also, you don't need a set of `foldMap` and `fold*Monoid` variants for each library that defines a `Monoid` type. Just pick one `Monoid` type, and define implicit conversions from `MonoidFromLibA` to whatever `Monoid` type you deem 'canonical' for this purpose. Basically, you the user can do exactly what you want FS2 to do. Right?

```scala
implicit class MonoidalFns[F[_],A](s: Stream[F,A]) {
  def foldMap[B:Monoidal](f: A => B): Stream[F,B]
}
object Monoidal {
  implicit def officalMonoidal[A:TehOfficialMonoid]: Monoidal[A] = ...
  implicit def anotherLibMonoidal[A:OtherLibMonoid]: Monoidal[A] = ...
}
trait Monoidal[A] { def monoid: TehOfficialMonoid[A] }
```